jvm-libs/linea/web3j-extensions/src/test/kotlin/linea/web3j/Web3JLogsSearcherIntTest.kt

```diff
@@ -0,0 +1,473 @@
+  ): SearchDirection? {
+    val number = ULong.fromHexString(ethLog.topics[1].encodeHex())
+    return when {
+      number < targetNumber -> SearchDirection.FORWARD
+      number > targetNumber -> SearchDirection.BACKWARD
+      else -> null
+    }
+  }
+
+  @Test
+  fun `findLogs searches and returns log when found`() {
+    setupClientWithTestingJsonRpcServer()
+
+    (100..200)
+      .forEach { number ->
+        logsClient.findLog(
+          fromBlock = 100UL.toBlockParameter(),
+          toBlock = 200UL.toBlockParameter(),
+          address = address,
+          topics = listOf("0xffaabbcc"),
+          chunkSize = 10,
+          shallContinueToSearchPredicate = { ethLog ->
+            shallContinueToSearch(ethLog, targetNumber = number.toULong())
+          }
+        )
+          .get()
+          .also { log ->
+            assertThat(log).isNotNull()
+            assertThat(log!!.topics[1].encodeHex()).isEqualTo(number.toULong().toHexStringUInt256())
+          }
+      }
+  }
+
+  @Test
+  fun `findLogs searches L1 and returns null when not found - before range`() {
+    setupClientWithTestingJsonRpcServer()
+
+    logsClient.findLog(
+      fromBlock = 100UL.toBlockParameter(),
+      toBlock = 200UL.toBlockParameter(),
+      address = address,
+      topics = listOf("0xffaabbcc"),
+      chunkSize = 10,
+      shallContinueToSearchPredicate = { ethLog ->
+        shallContinueToSearch(ethLog, targetNumber = 89UL)
+      }
+    )
+      .get()
+      .also { log ->
+        assertThat(log).isNull()
+        assertThat(TestingJsonRpcServer.callCountByMethod("eth_getLogs")).isBetween(1, 4)
+      }
+  }
+
+  @Test
+  fun `findLogs searches L1 and returns null when no logs in blockRange`() {
+    setupClientWithTestingJsonRpcServer(
+      subsetOfBlocksWithLogs = listOf(100UL..109UL, 150UL..159UL)
+    )
+
+    logsClient.findLog(
+      fromBlock = 100UL.toBlockParameter(),
+      toBlock = 200UL.toBlockParameter(),
+      address = address,
+      topics = listOf("0xffaabbcc"),
+      chunkSize = 10,
+      shallContinueToSearchPredicate = { ethLog ->
+        shallContinueToSearch(ethLog, targetNumber = 89UL)
+      }
+    )
+      .get()
+      .also { log ->
+        assertThat(log).isNull()
+      }
+  }
+
+  @Test
+  fun `findLogs searches L1 and returns null when not found - after range`() {
+    setupClientWithTestingJsonRpcServer()
+    logsClient.findLog(
+      fromBlock = 100UL.toBlockParameter(),
+      toBlock = 200UL.toBlockParameter(),
+      address = address,
+      topics = listOf("0xffaabbcc"),
+      chunkSize = 10,
+      shallContinueToSearchPredicate = { ethLog ->
+        shallContinueToSearch(ethLog, targetNumber = 250UL)
+      }
+    )
+      .get()
+      .also { log ->
+        assertThat(log).isNull()
+        assertThat(TestingJsonRpcServer.callCountByMethod("eth_getLogs")).isBetween(1, 4)
+      }
+  }
+
+  @Test
+  fun `findLogs searches L1 and returns null when range has no logs`() {
```

Comment:
What is the difference between `findLogs searches L1 and returns null when no logs in blockRange` and this scenario?

Reply:
yeah, the names are not great. 

the tests do 3 scenarios:
- target is before the range
- target is after the range
- target is middle of the range

I will rename for a more intuitive name

Reply:
Yeah there was some redundancy. I fixed it followup [PR](https://github.com/Consensys/linea-monorepo/pull/565). thank